Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/rapidantigen/RACoronaTest.kt

```diff
@@ -51,20 +52,30 @@ data class RACoronaTest(
     @SerializedName("dateOfBirth")
     val dateOfBirth: LocalDate?,
 
+    @SerializedName("outdatedAfter")
+    val outdatedAfter: Duration,
+
     @Transient override val isProcessing: Boolean = false,
     @Transient override val lastError: Throwable? = null,
 ) : CoronaTest {
 
     override val type: CoronaTest.Type = CoronaTest.Type.RAPID_ANTIGEN
 
-    fun getState(nowUTC: Instant) = when (testResult) {
-        CoronaTestResult.PCR_OR_RAT_PENDING -> State.PENDING
-        CoronaTestResult.RAT_NEGATIVE -> State.NEGATIVE
-        CoronaTestResult.RAT_POSITIVE -> State.POSITIVE
-        CoronaTestResult.RAT_INVALID -> State.INVALID
-        CoronaTestResult.RAT_REDEEMED -> State.REDEEMED
-        else -> throw IllegalArgumentException("Invalid RAT test state $testResult")
-    }
+    private fun isOutdated(nowUTC: Instant) = testedAt.plus(outdatedAfter).isBefore(nowUTC)
+
+    fun getState(nowUTC: Instant) =
+        if (isOutdated(nowUTC)) {
```

Comment:
The techspec graph says that a test can go from outdated to expired.

Reply:
hmm does it make a difference for the app? Id prefer to not make the state logic even more edgecasy

Reply:
the logic is now implemented as @mlenkeit suggested
`testResult == CoronaTestResult.PCR_NEGATIVE && isOutdated(nowUTC, testConfig) -> Outdated`
otherwise its based on the old state calculation